netty/src/main/java/io/grpc/netty/NettyClientTransport.java

```diff
@@ -211,6 +235,11 @@ public void shutdown() {
           = Status.UNAVAILABLE.withDescription("Channel requested transport to shut down");
       handler.getWriteQueue().enqueue(new GracefulCloseCommand(status), true);
     }
+    if (keepAliveManager != null) {
+      keepAliveManager.onTransportShutdown();
```

Comment:
Ah good call. Maybe we should rename it to `onTransportDisconnected` and clarify docs for it? It internally is using state named `DISCONNECTED`...